telegram/base.py

```diff
@@ -37,22 +36,21 @@
 class TelegramObject:
     """Base class for most Telegram objects."""
 
-    _id_attrs: Tuple[object, ...] = ()
-
     # Adding slots reduces memory usage & allows for faster attribute access.
     # Only instance variables should be added to __slots__.
-    # We add __dict__ here for backward compatibility & also to avoid repetition for subclasses.
-    __slots__ = ('__dict__',)
+    __slots__ = ('_id_attrs',)
+
+    def __new__(cls, *args: object, **kwargs: object) -> 'TelegramObject':  # pylint: disable=W0613
+        instance = super().__new__(cls)
+        instance._id_attrs = ()  # cannot add type hint since it is not read
```

Comment:
maybe rather "since mypy doesn't support __new__ yet"? I think it would also be good to add a comment on why we do this here in the first place (some for filters)

Reply:
>  "since mypy doesn't support __new__ yet"?

Actually it turns out it's a python thing

> it would also be good to add a comment on why we do this here in the first place (some for filters)

👌